File: programmers/연습문제/할인 행사.py
def solution(want, number, discount):
    answer = 0
    dic = {want[i]: number[i] for i in range(len(want))}
    days = len(discount)
    for i in range(days - 9):
        tmp = {}
        for p in discount[i: i + 10]:
            if p in tmp:
                tmp[p] += 1
            else:
                tmp[p] = 1
        if tmp == dic:
            answer += 1
    return answer

File: programmers/연습문제/할인 행사.js
function solution(want, number, discount) {
  function check(origin, candidate) {
    for (const item in origin) {
      if (origin[item] !== candidate[item]) return false;
    }
    return true;
  }
  let answer = 0;
  dic = {};
  for (let i = 0; i < want.length; i++) {
    dic[want[i]] = number[i];
  }
  for (let i = 0; i <= discount.length - 10; i++) {
    tmp = {};
    for (const item of discount.slice(i, i + 10)) {
      if (item in tmp) {
        tmp[item]++;
      } else {
        tmp[item] = 1;
      }
    }
    if (check(dic, tmp)) {
      answer++;
    }
  }
  return answer;
}
